lib/model/folder_sendrecv.go

```diff
@@ -1586,8 +1586,10 @@ func (f *sendReceiveFolder) dbUpdaterRoutine(dbUpdateChan <-chan dbUpdateJob) {
 				l.Debugf("fsync %q failed: %v", dir, err)
 				continue
 			}
-			if err := fd.Sync(); err != nil {
-				l.Debugf("fsync %q failed: %v", dir, err)
+			if !f.FolderConfiguration.DisableFsync {
```

Comment:
The entire flushfn shoudl go away if we're not syncing, and ideally we don't need to track changedDir either (but the cost isn't enormous so it might not be worth sprinkling yet more ifs to accomplish it, I didn't check)